### PR TITLE
Added add function, edit function and delete function using javascript.

### PR DESCRIPTION
**Pull request by zeff96**
Removed the hard-coded array objects and replaced them with the dynamic array.
Added add function to enable capture of data from the user input, store to local storage, and render it to the user on the page.

Added a function to allow the user to edit the list in case there is a need for,
Added a function to allow the user to delete the items from the list.

### CODE REVIEW

File: src/components/app.js [hunk 1]
```diff
@@ -1,36 +1,173 @@
...
+      list.isEditing = false;
+    }
+  });
+  saveItems();
+};
+
+const deleteItems = (listId) => {
+  lists = lists.filter((list) => list.index !== listId);
+  saveItems();
+};
```
Comment (tugsem) [inline src/components/app.js]: - [ ] Kindly create a new JavaScript file for adding, deleting and editing functionalities as requested in the project description:
image

File: src/components/app.js [hunk 2]
```diff
@@ -1,36 +1,173 @@
...
+  lists = savedItems;
+} else {
+  lists = [];
+}
+
+class Node {
+  constructor(description) {
+    this.description = description;
+    this.completed = false;
+    this.index = `${lists.length}`;
```
Comment (tugsem) [inline src/components/app.js]: - [ ] Your first item in the storage gets the index equals to 0. However, it should have the length of the new array which is 1. Kindly fix this.

File: src/components/app.js (hunk 1 above)
Comment (tugsem) [inline src/components/app.js]: - [ ]  This function should work for deleting single task/item. However, it doesn't work right now. Please consider naming it coherently and fix the error. Take attention to this functionality afterwards:
image

**Review by tugsem — CHANGES_REQUESTED** [review]
##  STATUS: REQUIRED CHANGES ♻️
Hi @zeff96 , 

Good job so far! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

Alt Text
## Highlights 🚀 
✔️ Gitflow used
✔️ Linters checked
✔️ App works properly
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by codecaiine — APPROVED** [review]
Hi @zeff96 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linters errors :heavy_check_mark: 
- The app works as expected :heavy_check_mark: 
 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**